tools/ratbagctl.in

```diff
@@ -1504,10 +1504,48 @@ def get_parser():
     return RatbagParserRoot(parser_def)
 
 
+def on_device_added(ratbagd, device):
+    device_names = [
+            'mara', 'capybara', 'porcupine', 'paca',
+            'vole', 'woodrat', 'gerbil', 'shrew',
+            'hutia', 'beaver', 'squirrel', 'chinchilla',
+            'mouse', 'rat', 'rabbit', 'guineapig',
+            'hare', 'degu', 'gundi', 'acouchy',
+            'nutria', 'tuco-tuco', 'paca', 'hamster',
+            'zokor', 'chipmunk', 'gopher', 'marmot',
+            'groundhog', 'suslik', 'agouti', 'blesmol',
+    ]
+
+    device_attr = [
+            'sobbing', 'whooping', 'barking', 'yapping',
+            'howling', 'hysterical', 'squawking', 'cheering',
+            'warbling', 'thundering', 'booming', 'blustering',
+            'humming', 'crying', 'bawling', 'shouting',
+            'roaring', 'raging', 'chanting', 'crooning',
+            'murmuring', 'bellowing', 'wailing', 'weeping',
+            'screaming', 'yelling', 'yodeling', 'singing',
+            'honking', 'hooting', 'whispering', 'hollering',
+    ]
+
+    # Let's convert the sha into something not boring. This takes the first
+    # 4 characters, creates two different indices from it to generate a
+    # name. The rest is hope hope that never get a collision here but it's
+    # unlikely enough.
+    name = device_names[int(device.id[0], 16) + int(device.id[1], 16)]
+    attr = device_attr[int(device.id[2], 16) + int(device.id[3], 16)]
```

Comment:
Hmm, this is not a proper random distribution. Here you are basically saying: "we take 2 dice (of 16), we roll them and get the sum of it to fetch the index". And that means you basically have much less chances of picking index `0` than `16` as `0 = 0 + 0` but `16 = 0 + 16 = 1 + 15 = 2 + 14 = ...` 

Assuming the device.id is at least 8 chars, we could do:
```
name = device_names[(int(device.id[0:2], 16) + int(device.id[2:4], 16)) % len(device_names)]
attr = device_attr[(int(device.id[4:6], 16) + int(device.id[6:8], 16)) % len(device_attr)]
```

This gives makes the distribution more flat as we are summing 2 dices that have a bigger size than each array.

Note that with that, in the first, third, fifth and seventh chars only the first bit will be taken into account (but does it really matters?)

Reply:
Fixed this by taking the first 4 bytes as 2-byte hex values and running a `% len(...)` on it. Also removed some of the entries I didn't like.